src/pkg/packager/deploy.go

```diff
@@ -109,6 +106,39 @@ func (p *Packager) Deploy() (err error) {
 	return nil
 }
 
+// attemptClusterChecks attempts to connect to the cluster and check for useful metadata and config mismatches.
+// NOTE: attemptClusterChecks should only return an error if there is a problem significant enough to halt a deployment, otherwise it should return nil and print a warning message.
+func (p *Packager) attemptClusterChecks() error {
+	// Connect to the cluster (if available) to check the Zarf Agent for breaking changes
+	var err error
+	if p.cluster, err = cluster.NewCluster(); err == nil {
+
+		// Get the init package from the cluster
+		if p.currentInitPackage, err = p.cluster.GetDeployedPackage("init"); err == nil {
```

Comment:
p.currentInitPackage isn't used outside of this function and should probably be a local variable.